test/artifact_bucket_test.go

```diff
@@ -0,0 +1,243 @@
+// can use a bucket for temporary storage of artifacts shared between tasks
+func TestStorageBucketPipelineRun(t *testing.T) {
+	configFilePath := os.Getenv("KANIKO_SECRET_CONFIG_FILE")
+	if configFilePath == "" {
+		t.Skip("KANIKO_SECRET_CONFIG_FILE variable is not set.")
+	}
+	logger := logging.GetContextLogger(t.Name())
+	c, namespace := setup(t, logger)
+
+	knativetest.CleanupOnInterrupt(func() { tearDown(t, logger, c, namespace) }, logger)
+	defer tearDown(t, logger, c, namespace)
+
+	bucketName := fmt.Sprintf("build-pipeline-test-%s-%d", namespace, time.Now().Unix())
+
+	logger.Infof("Creating Secret %s", bucketSecretName)
+	if _, err := c.KubeClient.Kube.CoreV1().Secrets(namespace).Create(getBucketSecret(t, logger, configFilePath, namespace)); err != nil {
+		t.Fatalf("Failed to create Secret `%s`: %s", bucketSecretName, err)
+	}
+	defer deleteBucketSecret(c, t, logger, namespace)
+
+	logger.Infof("Creating GCS bucket %s", bucketName)
+	createbuckettask := tb.Task("createbuckettask", namespace, tb.TaskSpec(
+		tb.TaskVolume("bucket-secret-volume", tb.VolumeSource(corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: bucketSecretName,
+			},
+		})),
+		tb.Step("step1", "google/cloud-sdk:alpine",
+			tb.Command("/bin/bash"),
+			tb.Args("-c", fmt.Sprintf("gcloud auth activate-service-account --key-file /var/secret/bucket-secret/bucket-secret-key && gsutil mb gs://%s", bucketName)),
+			tb.VolumeMount(corev1.VolumeMount{
+				Name:      "bucket-secret-volume",
+				MountPath: fmt.Sprintf("/var/secret/%s", bucketSecretName),
+			}),
+			tb.EnvVar("CREDENTIALS", fmt.Sprintf("/var/secret/%s/%s", bucketSecretName, bucketSecretKey)),
+		),
+	),
+	)
+
+	logger.Infof("Creating Task %s", "createbuckettask")
+	if _, err := c.TaskClient.Create(createbuckettask); err != nil {
+		t.Fatalf("Failed to create Task `%s`: %s", "createbuckettask", err)
+	}
+
+	createbuckettaskrun := tb.TaskRun("createbuckettaskrun", namespace,
+		tb.TaskRunSpec(tb.TaskRunTaskRef("createbuckettask")))
+
+	logger.Infof("Creating TaskRun %s", "createbuckettaskrun")
+	if _, err := c.TaskRunClient.Create(createbuckettaskrun); err != nil {
+		t.Fatalf("Failed to create TaskRun `%s`: %s", "createbuckettaskrun", err)
+	}
+
+	if err := WaitForTaskRunState(c, "createbuckettaskrun", TaskRunSucceed("createbuckettaskrun"), "TaskRunSuccess"); err != nil {
+		t.Errorf("Error waiting for TaskRun %s to finish: %s", "createbuckettaskrun", err)
+	}
+
+	defer runTaskToDeleteBucket(c, t, logger, namespace, bucketName, bucketSecretName, bucketSecretKey)
+
+	logger.Infof("Creating ConfigMap %s", v1alpha1.BucketConfigName)
+	configMapData := map[string]string{
+		v1alpha1.BucketLocationKey:              fmt.Sprintf("gs://%s", bucketName),
+		v1alpha1.BucketServiceAccountSecretName: bucketSecretName,
+		v1alpha1.BucketServiceAccountSecretKey:  bucketSecretKey,
+	}
+	c.KubeClient.UpdateConfigMap(systemNamespace, v1alpha1.BucketConfigName, configMapData)
```

Comment:
Just a small doubt. Are we updating config map after e2e test execution to its original state? Sorry if I missed this code

Reply:
adding login to reset config map to original values